dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -49,24 +60,115 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterF
 {
     auto lock = m_lock.lock_shared();
     RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
-    
+
+    // To be completed after Sharath's PR
+
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterForegroundActivator(_In_ IWpnForegroundSink* sink, _In_ PCWSTR processName)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterActivator(_In_ PCWSTR processName) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+    RETURN_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
+
+    std::wstring appId = GetAppIdentifier(processName);
+    m_notificationListenerManager.AddListener(appId, processName);
+
+    if (!m_notificationListenerManager.IsEmpty())
```

Comment:
This if check is not necessary. If the previous statement succeeded, then it can never be empty right?

Reply:
I looked at the documentation and there is no exception safety anywhere, so I assume if AddListener doesn't throw then we have at least one element in the listener manager.